API/Documentation.md

```diff
@@ -0,0 +1,37 @@
+
+## Essentials
+
+The following checklist lists documentation essentials. This is not an exhaustive list — APIs always have specifics which cannot be included in such lists, though you should document them too. A good rule of thumb is, if someone would ask about it, it should be documented.
+
+- **Overview** - Placed at the top of the documentation, the overview should explain: what the API does, architecture (e.g. REST), format (e.g. JSON:API or GraphQL) and should provide general information (base URL, response types, etc.).
+- **Format requirements** - If you use JSON:API, GraphQL or some other format, they typically come with requirements, like setting `Content-Type`/`Accept` headers, which the clients must abide by.
+- **CORS** - If applicable, document your CORS policy (allowed origins, methods, exposed headers, credentials, etc.).
+- **Authentication** - Applicable if endpoints are authenticated. Explain the authentication method(s) used (API key, cookies, JWT, etc.) and how to obtain credentials.
+- **Authorization** - If the outcome of requests depends on the user's permissions, then that should also be documented. If authorization logic varies between endpoints, each endpoint should expain its logic.
+- **Errors** - All APIs are subject to errors. Errors are roadblocks, preventing clients from doing what they need to do. And when a roadblock comes with either no explanation or a vague one (`PC LOAD LETTER`), frustration ensues.<br />
+Document <u>all</u> common errors and supply descriptions. Descriptions should explain what happened and what the client can do to remove the error. Document even the 5xx messages, no matter how obvious they seem.
+- **Changelog** - Major changes to the API should be documented for historical reference. It's an optional step if the API will be used by a limited number of users to whom you can communicate changes directly, though it's useful to have even then.
+- **Endpoints** - The main body of the documentation, each endpoint should: describe what it is, document available HTTP verbs, and show example requests and responses for happy and sad paths. If a response error is specific to an endpoint, document it. 
+
+*Bonus points:*
+- **Quickstart guide** - Onboard clients quickly by providing an example request (e.g. in cURL) which they can copy/paste to test out and modify. This approach is the fastest way to get clients to learn using the API.
+- **Diagrams** - Complex flows are hard to explain with just words. If the API, for example, controls a state machine, words will struggle to explain its states and transitions. Diagrams, even simple ones created in tools like [draw.io](https://draw.io), will help clients tremendously in understanding the underlying logic.<br />
+*Tip*: keep raw, editable files in the repository along with exported formats (like PNG). That way other developers can later update existing diagrams instead of having to create new ones.
+
```

Comment:
how about adding a section for "Direct upload flow" if application supports it?
Its a pretty complex flow and i think our frontend devs appreciate it.

Reply:
I'm 99% sure we had this chapter sometimes, somewhere...maybe it got transferred to devproc handbook?

Reply:
We have it mentioned [here](https://infinum.com/handbook/books/rails/development-practices/file-uploads#direct-s3-upload)

Reply:
Yeah, I think the one I'm thinking of had pictures, steps and such...but it's lost somewhere... 😅

Reply:
In the general section of the underline api documentation :D